libs/datamodel/core/src/ast/reformat/reformatter.rs

```diff
@@ -275,13 +276,28 @@ impl<'a> Reformatter<'a> {
         let mut block_name = "";
         let mut block_has_opened = false;
 
+        // sort directives
+        let directives = Self::extract_and_sort_directives(token, false);
+        let mut last_token_was_directive = false;
```

Comment:
This boolean looks like the hardest part to understand within this logic. This is because of the current treatment of `NEWLINE` tokens. I think you could quite easily adapt the definition of the `model_declaration` rule. Instead of `"@@" ~ directive` it might be good to use the `block_level_directive` rule which already contains the `NEWLINE`. This should enable you to remove this boolean altogether.

Reply:
That means in the parser a model like this would not be parsed correctly though:
```
model Blog {
  @@id([id1, id2]) @@map("blog")
  id1 Int
  id2 Int
}

Reply:
The parser would expect a newline after `@@id({id1, id2])`.

Reply:
I think that is fine. Your examples are the first one i saw that use it. Do we have test that explicitly states that this is a wanted feature?

Reply:
Got it, no I haven't found any tests using this.